src/hyperion/device_setup_plans/check_topup.py

```diff
@@ -0,0 +1,72 @@
+import bluesky.plan_stubs as bps
+from dodal.devices.detector import DetectorParams
+from dodal.devices.synchrotron import Synchrotron, SynchrotronMode
+
+from hyperion.log import LOGGER
+
+ALLOWED_MODES = [SynchrotronMode.USER, SynchrotronMode.SPECIAL]
+
+
+def _in_decay_mode(time_to_topup):
+    if time_to_topup == -1:
```

Comment:
This feels like a bit of a magic number. From context I'm guessing it's what the synchrotron PV reports when the synchrotron is in decay mode, maybe naming this constant would make it clearer.